contracts/DXswapFeeReceiver.sol

```diff
@@ -0,0 +1,127 @@
+pragma solidity =0.5.16;
+
+import './interfaces/IDXswapFactory.sol';
+import './interfaces/IDXswapPair.sol';
+import './interfaces/IWETH.sol';
+import './libraries/TransferHelper.sol';
+import './libraries/SafeMath.sol';
+
+
+contract DXswapFeeReceiver {
+    using SafeMath for uint;
+
+    address public owner;
+    IDXswapFactory public factory;
+    address public WETH;
+    address public ethReceiver;
+    address public fallbackReceiver;
+
+    constructor(
+        address _owner, address _factory, address _WETH, address _ethReceiver, address _fallbackReceiver
+    ) public {
+        owner = _owner;
+        factory = IDXswapFactory(_factory);
+        WETH = _WETH;
+        ethReceiver = _ethReceiver;
+        fallbackReceiver = _fallbackReceiver;
+    }
+    
+    function() external payable {}
+
+    function transferOwnership(address newOwner) external {
+        require(msg.sender == owner, 'DXswapFeeReceiver: FORBIDDEN');
+        owner = newOwner;
+    }
+    
+    function changeReceivers(address _ethReceiver, address _fallbackReceiver) external {
+        require(msg.sender == owner, 'DXswapFeeReceiver: FORBIDDEN');
+        ethReceiver = _ethReceiver;
+        fallbackReceiver = _fallbackReceiver;
+    }
+    
+    // Returns sorted token addresses, used to handle return values from pairs sorted in this order
+    function sortTokens(address tokenA, address tokenB) internal pure returns (address token0, address token1) {
+        require(tokenA != tokenB, 'DXswapFeeReceiver: IDENTICAL_ADDRESSES');
+        (token0, token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+        require(token0 != address(0), 'DXswapFeeReceiver: ZERO_ADDRESS');
+    }
+    
+    // Helper function to know if an address is a contract
+    function isContract(address addr) internal returns (bool) {
+        uint size;
+        assembly { size := extcodesize(addr) }
+        return size > 0;
+    }
+
+    // Calculates the CREATE2 address for a pair without making any external calls
+    // Taken from DXswapRouter, removed the factory parameter
```

Comment:
This is used in the DXswapRouter, but the code lives here: https://github.com/levelkdev/dxswap-periphery/blob/master/contracts/libraries/DXswapLibrary.sol#L18